### ISSUE TITLE
what is the default tokenizer when "No sentence-transformers model found with name"?

### ISSUE THREAD

**Issue by Hortatori** [UKPLab/sentence-transformers#2779]
I'm trying to use the sentence-transformer dangvantuan/sentence-camembert-large model and I'm getting a "no model found" error. This error is probably because some Sentence-Transformers-specific files are missing in their Huggingface (modules.json and config_sentence_transformers.json). 
But then, Sentence Transformer warns it will create a new model with mean pooling, and this model performs really well on my data (!). 
So, I would like to know what the tokeniser's model is when the model name hasn't been found?

**Comment by tomaarsen** [comment]
Hello!

https://huggingface.co/dangvantuan/sentence-camembert-large indeed doesn't have any SentenceTransformer-specific files (which is a bit odd, as it does recommend using Sentence Transformers in their Usage), so it indeed "creates a new model with mean pooling". I'd like to elaborate a bit on that.

Loading a model goes one of 2 ways:
1. Read the `modules.json` file which contains information on each of the modules that make up a ST model. The first one is often a `Transformer` module, and this one will be used to load the model. Most of the time, the second one is a `Pooling` module with some settings, and then sometimes there's a `Normalize` module as well.
2. If the previous fails because there's no `modules.json`, then we take some default set of modules: a `Transformer` and a `Pooling` with mode="mean". So, the model loading is actually very similar as if there was a `modules.json`.

The `Transformer` module will internally call:
```python
model = AutoModel.from_pretrained(model_name)
tokenizer = AutoTokenizer.from_pretrained(model_name)
```
Regardless of whether the model loading goes "route 1 or route 2". 

In short: even with a `will create a new model with mean pooling` warning, your model uses the tokenizer from https://huggingface.co/dangvantuan/sentence-camembert-large.

Based on https://huggingface.co/dangvantuan/sentence-camembert-large#usage, it seems like you're using the model as it's intended.

- Tom Aarsen

**Comment by Hortatori** [comment]
Thanks a lot!
